app/services/sirsi_response/error.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class SirsiResponse::Error
+  attr_reader :html, :log
+
+  SIRSI_RESPONSE_TRANSLATIONS = {
+    "hatErrorResponse.141": 'Denied: Renewal limit reached, cannot be renewed.',
+    "hatErrorResponse.7703": 'Denied: Renewal limit reached, cannot be renewed.',
+    "hatErrorResponse.105": 'Denied: Item has been recalled, cannot be renewed.',
+    "hatErrorResponse.252": 'Denied: Item has holds, cannot be renewed.',
+    "hatErrorResponse.46": 'Denied: Item on reserve, cannot be renewed.',
+    "unhandledException": 'Denied: Item cannot be renewed.'
+  }.with_indifferent_access
+
+  def initialize(error_message_raw:, symphony_client:, key:, session_token:, bib_type:)
+    @log = error_message_raw
+    display_error = make_human_friendly_response
+    parsed_response = wrangle_symphony bib_type, symphony_client, session_token, key
+    klass = bib_type.to_s.classify.constantize
+    bib_record = klass.new parsed_response
+
+    @html = RedisJobsController.render template: 'sirsi_response/error',
+                                       layout: false,
+                                       locals: { id: key,
+                                                 title: bib_record.title,
+                                                 display_error: display_error }
+  end
+
+  private
+
+    def make_human_friendly_response
+      SIRSI_RESPONSE_TRANSLATIONS[@log&.dig('messageList')&.first&.dig('code')] ||
```

Comment:
these translations are only for renewals, is it like that here? @cdmo

Reply:
No, it's for any errors, but it doesn't hurt to check. I imagine we may find more in the future in holds requests? Or if other errors showup. If not we can add a check for bib type

Reply:
Afair we were gonna keep the other error messages as the way sirsi has, currently the cat only have these for renewals, we can check with Ruth, if so adding a check for bib type makes sense.

Reply:
Okay, that sounds good. Thanks. Guess we could also abstract this Error class a bit more and then make subclasses for CheckoutError and HoldError.

Reply:
checking with Ruth https://github.com/psu-libraries/myaccount/issues/179#issuecomment-653581309

Reply:
I think it's nice to have the option to override whenever we want, and doing this doesn't cost much and keeps things simpler. 🤷